cmd/snap/cmd_snap_op.go

```diff
@@ -579,6 +580,23 @@ func (x *cmdTry) Execute([]string) error {
 		JailMode: x.JailMode,
 	}
 
+	if name == "" {
+		if _, err := os.Stat("snapcraft.yaml"); err == nil {
+			if stat, err := os.Stat("prime"); err == nil && stat.IsDir() {
+				name = "prime"
+			}
+		} else {
+			if stat, err := os.Stat("meta"); err == nil && stat.IsDir() {
+				if _, err := os.Stat("meta/snap.yaml"); err == nil {
+					name = "./"
+				}
+			}
+		}
+		if name == "" {
+			return fmt.Errorf(i18n.G("error: the `<snap-dir>` argument was not provided and neither `prime` nor `meta` directory could be found"))
```

Comment:
I'd suggest `"[...] argument was not provided and couldn't be inferred"`, with the inference mechanism explained in the long help.